addons/partner_autocomplete/models/res_partner.py

```diff
@@ -0,0 +1,309 @@
+            'child_ids': self._format_data_company_contact(company_data.get('contacts')) if company_data.get('contacts') else [],
+        }
+
+        return company
+
+    @api.model
+    def _find_country_data(self, state_code, state_name, country_code, country_name):
+        result = {
+            'country_id': False,
+            'state_id': False
+        }
+
+        country_id = self.env['res.country'].search([['code', '=ilike', country_code]])
+        if not country_id:
+            country_id = self.env['res.country'].search([['name', '=ilike', country_name]])
+
+        if country_id:
+            result['country_id'] = {
+                'id': country_id.id,
+                'display_name': country_id.display_name
+            }
+            if state_name or state_code:
+                state_id = self.env['res.country.state'].search([
+                    ('country_id', '=', country_id.id),
+                    '|',
+                    ('name', '=ilike', state_name),
+                    ('code', '=ilike', state_code)
+                ], limit=1)
+
+                if state_id:
+                    result['state_id'] = {
+                        'id': state_id.id,
+                        'display_name': state_id.display_name
+                    }
+
+        else:
+            _logger.info('Country code not found: %s', country_code)
+
+        return result
+
+    @api.model
+    def _rpc_remote_api(self, action, params, timeout=15):
+        url = '%s/%s' % (PARTNER_REMOTE_URL, action)
+        account = self.env['iap.account'].get('partner_autocomplete')
+        params.update({
+            'db_uuid': self.env['ir.config_parameter'].sudo().get_param('database.uuid'),
+            'account_token': account.account_token,
+            'country_code': self.env.user.company_id.country_id.code,
+            'zip': self.env.user.company_id.zip,
+        })
+        try:
+            return jsonrpc(url=url, params=params, timeout=timeout), False
+        except (ConnectionError, HTTPError, exceptions.AccessError) as exception:
+            _logger.error('Autocomplete API error: %s' % str(exception))
+            return False, str(exception)
+
+    @api.model
+    def autocomplete(self, query):
+        suggestions, error = self._rpc_remote_api('search', {
+            'query': query,
+        })
+        if suggestions:
+            results = []
+            for suggestion in suggestions:
+                results.append(self._format_preview_data(suggestion))
+            return results
+        else:
+            return []
+
+    @api.model
+    def enrich_company(self, company_domain, company_data_id, vat):
+        response, error = self._rpc_remote_api('enrich', {
+            'domain': company_domain,
+            'company_data_id': company_data_id,
+            'vat': vat,
+        })
+        if response and response.get('company_data'):
```

Comment:
d =  response.get('company_data'):
# map country and state to ids
d._map_country_and_sate_to_ids:
d['additional_info'] = self.ref('partner_autocmploete.info_template').render(d)

return